src/overlays/actors/ovl_En_Nutsball/z_en_nutsball.c

```diff
@@ -23,12 +26,151 @@ const ActorInit En_Nutsball_InitVars = {
 };
 */
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Nutsball_0x80985C40/EnNutsball_Init.asm")
+extern ColliderCylinderInit D_809861F0;
+/*
+static ColliderCylinderInit sCylinderInit = {
+    { 
+        COLTYPE_NONE, 
+        AT_ON | AT_TYPE_ENEMY, 
+        AC_ON | AC_TYPE_PLAYER, 
+        OC1_ON | OC1_TYPE_ALL, 
+        OC2_TYPE_2, 
+        COLSHAPE_CYLINDER, 
+    },
+    { 
+        ELEMTYPE_UNK0, 
+        { 0xF7CFFFFF, 0x00, 0x04 }, 
+        { 0xF7CFFFFF, 0x00, 0x00 }, 
+        TOUCH_ON | TOUCH_SFX_WOOD, 
+        BUMP_ON, 
+        OCELEM_ON, 
+    },
+    { 13, 13, 0, { 0, 0, 0 } },
+};
+*/
+
+void EnNutsball_Init(Actor *thisx, GlobalContext *globalCtx) {
+    EnNutsball *this = THIS;
+    
+    ActorShape_Init(&this->actor.shape, 400.0f, (ActorShadowFunc)func_800B3FC0, 13.0f);
+    Collider_InitAndSetCylinder(globalCtx, &this->collider, &this->actor, &D_809861F0);
+    this->actor.shape.rot.y = 0;
+    this->actor.speedXZ = 10.0f;
+    if (this->actor.params == 2) {
+        this->timer = 1;
+        this->unk146 = 0;
+        this->actor.gravity = -1.0f;
+    } else {
+        this->timer = 20;
+        this->unk146 = 19;
+        this->actor.gravity = -0.5f;
+    }
+    this->actor.world.rot.x = -this->actor.shape.rot.x;
+    this->actor.shape.rot.x = 0;
+    if (this->actor.params == 1) {
+        func_80985D3C(this);
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Nutsball_0x80985C40/EnNutsball_Destroy.asm")
+void EnNutsball_Destroy(Actor *thisx, GlobalContext *globalCtx) {
+    EnNutsball *this = THIS;
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Nutsball_0x80985C40/func_80985D3C.asm")
+void func_80985D3C(EnNutsball *this) {
+    this->collider.base.atFlags &= ~AT_TYPE_ENEMY & ~AT_BOUNCED & ~AT_HIT;
+    this->collider.base.atFlags |= AT_TYPE_PLAYER;
+    this->collider.info.toucher.dmgFlags = 0x400000;
+    this->collider.info.toucher.damage = 2;
+}
 
+#if NON_MATCHING
+//sp44 is at 0x48(sp) instead of 0x44(sp) while sp40 is at 0x44(sp) instead of 0x40(sp) 
+void EnNutsball_Update(Actor *thisx, GlobalContext *globalCtx) {
+    EnNutsball *this = THIS;
+    GlobalContext *globalCtx2 = globalCtx;
+    
+    ActorPlayer *player = PLAYER;
+    Vec3f sp60;
+    Vec3s sp58;
+    Vec3f sp4c;
+    u32 sp44;
+    CollisionPoly *sp40;
+
+    if (!(player->stateFlags1 & 0x300000C0)) {
+        this->timer--;
+        if (this->timer < 0) {
+            this->actor.velocity.y += this->actor.gravity;
+            this->actor.world.rot.x = Math_FAtan2F(sqrtf((this->actor.velocity.x * this->actor.velocity.x) + (this->actor.velocity.z * this->actor.velocity.z)), this->actor.velocity.y);
```

Comment:
Don't do all the calculations on one line, instead use the temp.
```suggestion
            temp = sqrtf((this->actor.velocity.x * this->actor.velocity.x) + (this->actor.velocity.z * this->actor.velocity.z));
            this->actor.world.rot.x = Math_FAtan2F(temp, this->actor.velocity.y);
```

Reply:
Thanks for the suggestion